std/uri.d

```diff
@@ -372,6 +372,47 @@ string encodeComponent(Char)(in Char[] uriComponent) if (isSomeChar!Char)
     return URI_Encode(s, URI_Alpha | URI_Digit | URI_Mark);
 }
 
+/* Encode associative array using www-form-urlencoding
+ *
+ * Params:
+ *      values = an associative array containing the values to be encoded.
+ *
+ * Returns:
+ *      A string encoded using www-form-urlencoding.
+ */
+package string urlEncode(int reserve = -1)(in string[string] values)
```

Comment:
the template parameter is just overkill here. For every value a new template will be instantiated and additionally using -1 as a default value is not really D style. I'm also not really sure why you would manually reserve.